### ISSUE TITLE
Use other kind of object different to continuous, or categorical

### ISSUE THREAD

**Issue by Afr063426** [bethatkinson/rpart#69]
Hello, I am trying to use my own metric to define splits. The y variable is no continuous neither categorical. It is an object created, and it is a list with two vectors with `vctrs::new_vctr.` I get this error "Error in model.frame.default(formula = histograma ~ prueba, data = histogramas_ejemplo  :  invalid type (list) for variable 'histograma'" Someone have tried something similar? And how do you solve this?

**Comment by bethatkinson** [comment]
The function is designed to use results from model.matrix().  Does the endpoint that you've created work with model.matrix?

From: Joshua Cervantes Artavia ***@***.***>
Sent: Monday, October 14, 2024 9:12 PM
To: bethatkinson/rpart ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] [bethatkinson/rpart] Use other kind of object different to continuous, or categorical (Issue #69)


Hello, I am trying to use my own metric to define splits. The y variable is no continuous neither categorical. It is an object created, and it is a list with two vectors with vctrs::new_vctr. I get this error "Error in model.frame.default(formula = histograma ~ prueba, data = histogramas_ejemplo : invalid type (list) for variable 'histograma'" Someone have tried something similar? And how do you solve this?

-

**Comment by Afr063426** [comment]
No, should I do that work with `model.matrix`? And with this should it work?

**Comment by bethatkinson** [comment]
Start by getting model.matrix to work

From: Joshua Cervantes Artavia ***@***.***>
Sent: Wednesday, October 16, 2024 7:52 AM
To: bethatkinson/rpart ***@***.***>
Cc: Atkinson, Beth J., M.S. ***@***.***>; Comment ***@***.***>
Subject: [EXTERNAL] Re: [bethatkinson/rpart] Use other kind of object different to continuous, or categorical (Issue #69)


No, should I do that work with model.matrix? And with this should it work?

-

**Comment by Afr063426** [comment]
Ok thank you so much!
